### DISCUSSION TITLE
How to run Llama3-8b instruct model on multiple GPUs?

### DISCUSSION THREAD

**Discussion by aitechguy0105** [ggml-org/llama.cpp#7086] [General]
Deos llama.cpp support llama3-8b to run on multiple GPUs?

**Comment by gardner** [comment]
Why would you run it on multiple GPUs?

**Comment by aitechguy0105** [comment]
to test benchmark. other models seems to be supported on multiple GPUs.

**Reply by gardner** [reply]
It only actually runs inference on one GPU at a time. You get the benefit of the additional VRAM but not the benefit of the additional processing power. Llama 3 8B should work on multiple GPUs. Your question doesn't really provide a lot of context. To maximize the possibility of someone being able to understand your issue and to help you, please review this article on [How To Ask Questions The Smart Way](http://www.catb.org/~esr/faqs/smart-questions.html).

**Comment by aknirala** [comment]
One reason for trying to run on multiple GPUs, as in my case is, it does not fit on one. I have 2 Nividia 2080 tI, each has around 11GB of memory. By default it choses GPU:0. I was wondering if I can parallelize between the two GPUs to run it.

**Reply by gardner** [reply]
The Q4_K_M quant is about 4.92 GB. It should fit in 11 GB of VRAM with space left for inference.

It sounds like you might be interested in this discussion: https://github.com/ggerganov/llama.cpp/discussions/8725
